src/__test__/components/plots-and-tables/markerHeatmap.test.jsx

```diff
@@ -0,0 +1,121 @@
+import rootReducer from '../../../redux/reducers/index';
+import genes from '../../../redux/reducers/genes/initialState';
+import * as loadConfig from '../../../redux/reducers/componentConfig/loadConfig';
+import HeatmapPlot from '../../../pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index';
+import { fetchCachedWork } from '../../../utils/cacheRequest';
+import { mockCellSets1 as cellSets } from '../../test-utils/cellSets.mock';
+
+jest.mock('localforage');
+enableFetchMocks();
+jest.mock('../../../components/plots/Header', () => () => <div />);
+
+jest.mock('../../../utils/cacheRequest', () => ({
+  fetchCachedWork: jest.fn().mockImplementation((expId, body) => {
+    if (body.name === 'ListGenes') {
+      return new Promise((resolve) => resolve({
+        rows: [{ gene_names: 'MockGeneWithHighestDispersion', dispersions: 54.0228 }],
+      }));
+    }
+    if (body.name === 'MarkerHeatmap') {
+      return new Promise((resolve) => {
+        resolve('resolved');
+      });
+    }
+  }),
+}));
+const defaultStore = {
+  cellSets,
+  componentConfig: initialComponentConfigStates,
+  embeddings: {},
+  experimentSettings: {
+    ...initialExperimentState,
+    backendStatus: {
+      status: {
+        pipeline: {
+          startDate: '2020-01-01T00:00:00',
+          status: 'SUCEEDED',
+        },
+        worker: { status: 'Running' },
+        gem2s: { status: 'SUCCEEDED' },
+      },
+    },
+    processing: {
+      configureEmbedding: {
+        clusteringSettings: {
+          methodSettings: {
+            louvain: {
+              resolution: 0.8,
+            },
+          },
+        },
+      },
+    },
+
+  },
+  genes,
+};
+
+const experimentId = 'randomExperiment';
+let store = null;
+let loadConfigSpy = null;
+let loadMarkersSpy;
+
+const renderHeatmapPage = async () => {
+  store = createStore(rootReducer, _.cloneDeep(defaultStore), applyMiddleware(thunk));
+  rtl.render(
+    <Provider store={store}>
+      <HeatmapPlot
+        experimentId={experimentId}
+      />
+    </Provider>,
+  );
+  await rtl.waitFor(() => expect(loadConfigSpy).toHaveBeenCalledTimes(1));
+  await rtl.waitFor(() => expect(fetchCachedWork).toHaveBeenCalledTimes(1));
+};
+
+describe('Marker heatmap plot', () => {
+  beforeAll(async () => {
+    await preloadAll();
+  });
+  beforeEach(() => {
+    jest.clearAllMocks(); // Do not mistake with resetAllMocks()!
+    fetchMock.resetMocks();
+    fetchMock.doMock();
+    fetchMock.mockResponse(JSON.stringify({}), { status: 404, statusText: '404 Not Found' });
+    loadConfigSpy = jest.spyOn(loadConfig, 'default');
+    loadMarkersSpy = jest.spyOn(markerGenesLoaded, 'default');
+  });
+
+  // marker heatmap renders
+  it('loads initially', async () => {
+    await renderHeatmapPage();
+
+    expect(loadMarkersSpy).toHaveBeenCalled();
+  });
+
+  it('loads marker genes on selecting', async () => {
```

Comment:
we need more tests apart from loading marker genes on selecting. 
Tests like "the plot changes on change of a given schema parameter", checking that the schema is updated correctly, checking that the relevant update calls were triggered.
Tests about how would the marker heatmap behave if a given part of the redux store is not defined: what if genes were missing? What if experimentSettings were undefined, etc. 
Try to think about what can possibly go wrong and add tests to check about it.